### PR TITLE
replica: rename table::on_compaction_completion and coroutinize it

### PR DESCRIPTION
**Pull request by raphaelsc**
on_compaction_completion() is not very descriptive. let's rename
it, following the example of
update_sstable_lists_on_off_strategy_completion().

Also let's coroutinize it, so to remove the restriction of running
it inside a thread only.

Signed-off-by: Raphael S. Carvalho <raphaelsc@scylladb.com>

### CODE REVIEW

File: replica/table.cc [hunk 1]
```diff
@@ -961,8 +961,8 @@ table::update_sstable_lists_on_off_strategy_completion(sstables::compaction_comp
 }
 
 // Note: must run in a seastar thread
```
Comment (denesb) [inline replica/table.cc:963]: This comment is now stale, no?

**Review by denesb — APPROVED** [review]
LGTM, one minor comment.

File: replica/database.hh [hunk 2]
```diff
@@ -589,7 +589,7 @@ private:
     update_sstable_lists_on_off_strategy_completion(sstables::compaction_completion_desc desc);
 
     // Rebuild sstable set, delete input sstables right away, and update row cache and statistics.
-    void on_compaction_completion(sstables::compaction_completion_desc desc);
+    future<> update_main_sstable_list_on_compaction_completion(sstables::compaction_completion_desc desc);
```
Comment (avikivity) [inline replica/database.hh:592]: Absolutely approve of those super long names. Rare and complicated events need long and descriptive names.

File: replica/table.cc (hunk 1 above)
Comment (raphaelsc) [reply replica/table.cc:963]: Yes it is. Will fix and resend.

File: replica/database.hh (hunk 2 above)
Comment (raphaelsc) [reply replica/database.hh:592]: > Absolutely approve of those super long names. Rare and complicated events need long and descriptive names.

Agreed.

**Comment by raphaelsc** [comment]
v2: Remove stale comment

**Pull request closed without merging**